src/caja-desktop-window.c

```diff
@@ -289,17 +368,23 @@ realize (GtkWidget *widget)
     CajaDesktopWindowPrivate *details;
     window = CAJA_DESKTOP_WINDOW (widget);
     details = window->details;
+    GdkDisplay *display;
 
     /* Make sure we get keyboard events */
-    gtk_widget_set_events (widget, gtk_widget_get_events (widget)
-                           | GDK_KEY_PRESS_MASK | GDK_KEY_RELEASE_MASK);
-    /* Do the work of realizing. */
-    GTK_WIDGET_CLASS (caja_desktop_window_parent_class)->realize (widget);
+    if (GDK_IS_X11_WINDOW (GDK_WINDOW(widget)))
```

Comment:
This is not right, `widget` is not a `GdkWindow`.  And as you use the display a little bit down, just use that here as well

Reply:
Tossing that, reverting to the display check